### PR TITLE
Add converted high dimensional covariance testing

### PR DESCRIPTION
**Pull request by IlhaH**
Added initial attempts for converted two sample test functions for covariance matrices, along with tests and a data file to compare original outputs with

## Summary by CodeRabbit

- **New Features**
  - Introduced advanced hypothesis testing and statistical analysis tools, including two-sample tests, eigenvalue computations, and kernel-based calculations.
  - Added functions for various statistical tests (CLX2013, SY2010, HC2018) and adaptive step-up procedures for multiple hypothesis testing.
  - Enhanced capabilities for rigorous statistical analyses across diverse scenarios.

### CODE REVIEW

File: python/python/bystro/covariance/hypothesis_hd.py [hunk 1]
```diff
@@ -0,0 +1,873 @@
...
+    -------
+    tuning_result : dict
+        Tuned constant C and rates.
+    """
+    rng = np.random.default_rng(seed)
+    Cs = np.arange(1, 31) / 10
+    rates = []
+
+    if thres is None:
+        thres = 2.6
```
Comment (akotlar) [inline python/python/bystro/covariance/hypothesis_hd.py:434]: Why do we set this default? Could we logger.warn("Threshold `thres` not specified, setting %d", 2.6)?

File: python/python/bystro/covariance/hypothesis_hd.py [hunk 2]
```diff
@@ -0,0 +1,873 @@
...
+    Xs, Ys, Zs = sample_list["Xs"], sample_list["Ys"], sample_list["Zs"]
+
+    eig_xs = cov_eigs(Xs)
+    eig_ys = cov_eigs(Ys)
+    eig_zs = cov_eigs(Zs)
+    gamma = np.median(eig_zs)
+    if not check_efficient(gamma, eig_xs, eig_ys, epsilon):
+        return {"efficient": False, "c": 1}
+
+    eta0 = np.std(eig_zs, ddof=1) * const
```
Comment (akotlar) [inline python/python/bystro/covariance/hypothesis_hd.py:203]: Could we also validate that const, alpha, epsilon, thres are in valid ranges? Typically we have been raising ValueError if an input is incorrect (e.g., `r("Invalid input: const must be a float that is greater than x")`

**Comment by akotlar** [comment]
OK by me!

**Comment by austinTalbot7241993** [comment]
LGTM

**Review by austinTalbot7241993 — APPROVED** [review]

**Pull request merged**